File: app/screens/InputWeightScreen.js
import React, { useState } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Icon from 'react-native-vector-icons/Ionicons';
import ScrollPicker from "react-native-wheel-scrollview-picker";

const WeightInputScreen = () => {
  const navigation = useNavigation();
  const [selectedWeight, setSelectedWeight] = useState(40); // Initial weight set to 31
  
  // Adjusted weightOptions to start from 20 and go up to 240
  const weightOptions = Array.from({ length: 240 - 20 + 1 }, (_, index) => index + 20);

  return (
    <View style={styles.container}>
      <TouchableOpacity style={styles.arrow} onPress={() => navigation.goBack()}>
        <Icon name="arrow-back" size={24} color="black" />
      </TouchableOpacity>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>What's your weight(kg)?</Text>
      </View>
      <View style={styles.scrollPickerContainer}>
        <ScrollPicker
          dataSource={weightOptions}
          selectedIndex={selectedWeight - 20} // Adjust index to match the new range starting at 20
          renderItem={(data, index) => (
            <Text style={styles.pickerItem}>{data}</Text>
          )}
          onValueChange={(data, index) => setSelectedWeight(data)} // Use the actual selected weight
          wrapperHeight={180}
          wrapperBackground="#FFFFFF"
          itemHeight={35}
          highlightColor="#000"
          highlightBorderWidth={2}
        />
      </View>
      <TouchableOpacity style={styles.button} onPress={() => {
        // Handle continue action, e.g., navigation to next screen
        if (selectedWeight) {
          console.log('Weight of the user: ', selectedWeight)
          // Navigate to the next screen with the selected weight
          navigation.navigate('StatusPage', { weight: selectedWeight });
        }
      }}>
        <Text style={styles.buttonText}>Continue</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
    backgroundColor: '#fff'
  },
  arrow: {
    position: 'absolute',
    left: '8%',
    top: '8%',
  },
  titleContainer: {
    position: 'absolute',
    top: '20%', 
    width: '100%',
    alignItems: 'center',
  },
  title: {
    fontSize: 28,
    color: '#0000ff',
  },
  scrollPickerContainer: {
    position: 'absolute',
    top: '40%',
    height: 180,
  },
  pickerItem: {
    textAlign: 'center',
    alignItems: 'center',
  },
  button: {
    backgroundColor: '#0000ff',
    paddingVertical: 15,
    paddingHorizontal: 80,
    borderRadius: 25,
    position: 'absolute',
    bottom: 30,
  },
  buttonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
  },
});

export default WeightInputScreen;

File: flutter/lib/inputweight_screen.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';
import 'package:flutter_secure_storage/flutter_secure_storage.dart';

class WeightInputScreen extends StatefulWidget {
  @override
  _WeightInputScreenState createState() => _WeightInputScreenState();
}

class _WeightInputScreenState extends State<WeightInputScreen> {
  int selectedWeight = 60; // Default weight
  List<int> weightOptions = List.generate(
      200, (index) => index + 30); // Weight options from 30 to 229

  final storage = FlutterSecureStorage(); // Secure storage instance

  Future<void> _validateAndProceed() async {
    print('Selected weight: $selectedWeight');

    // Retrieve JWT token from Flutter Secure Storage
    String? token = await storage.read(key: 'jwt_token');
    if (token == null) {
      print("Token not found. Please log in again.");
      // Handle token missing (perhaps navigate to login)
      return;
    }

    // Call the Flask backend with the selected weight
    final response = await http.post(
      Uri.parse('http://192.168.149.8:8081/submit_weight'), // Your backend URL
      headers: {
        'Authorization':
            'Bearer $token', // Pass JWT token in Authorization header
        'Content-Type': 'application/json',
      },
      body: json.encode({'weight': selectedWeight}),
    );

    if (response.statusCode == 200) {
      // Handle success
      print('Weight submitted successfully');
      Navigator.pushNamed(
          context, '/status'); // Example: navigate to the next screen
    } else {
      // Handle error
      print('Failed to submit weight: ${response.body}');
      // Show error message to the user
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: Column(
        children: [
          // Custom back button
          Padding(
            padding: const EdgeInsets.only(top: 30.0, left: 10.0),
            child: Align(
              alignment: Alignment.topLeft,
              child: IconButton(
                icon: Icon(Icons.arrow_back, color: Colors.black),
                onPressed: () => Navigator.pop(context),
              ),
            ),
          ),
          // Title
          Padding(
            padding: const EdgeInsets.only(top: 60.0, bottom: 10.0),
            child: Text(
              'What is your weight?',
              style: TextStyle(
                fontSize: 28,
                color: Colors.blue,
                fontWeight: FontWeight.bold,
              ),
              textAlign: TextAlign.center,
            ),
          ),

          // Weight Picker
          Expanded(
            child: Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                _buildPicker(weightOptions, (index) {
                  setState(() {
                    selectedWeight = weightOptions[index];
                  });
                }, weightOptions.indexOf(selectedWeight)),
              ],
            ),
          ),

          // Continue Button
          Padding(
            padding: const EdgeInsets.only(bottom: 50.0),
            child: ElevatedButton(
              style: ElevatedButton.styleFrom(
                backgroundColor: Colors.blue,
                padding: EdgeInsets.symmetric(vertical: 15, horizontal: 80),
                shape: RoundedRectangleBorder(
                  borderRadius: BorderRadius.circular(25),
                ),
              ),
              onPressed: _validateAndProceed,
              child: Text(
                'Continue',
                style: TextStyle(
                  color: Colors.white,
                  fontSize: 18,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildPicker(List<dynamic> items, Function(int) onSelectedItemChanged,
      int initialItem) {
    return SizedBox(
      width: 120,
      height: 135,
      child: Stack(
        alignment: Alignment.center,
        children: [
          // Highlight lines for the selected item
          Container(
            height: 45,
            decoration: BoxDecoration(
              border: Border(
                top: BorderSide(color: Colors.blue, width: 2.0),
                bottom: BorderSide(color: Colors.blue, width: 2.0),
              ),
            ),
          ),
          // ListWheelScrollView to show weight options
          ListWheelScrollView.useDelegate(
            itemExtent: 45,
            diameterRatio: 1.5,
            physics: FixedExtentScrollPhysics(),
            onSelectedItemChanged: onSelectedItemChanged,
            controller: FixedExtentScrollController(initialItem: initialItem),
            childDelegate: ListWheelChildBuilderDelegate(
              builder: (context, index) {
                return Center(
                  child: Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: [
                      Text(
                        '${items[index]}', // Show weight value
                        style: TextStyle(
                          fontSize: 20,
                          fontWeight: FontWeight.bold,
                          color: Colors.black,
                        ),
                      ),
                    ],
                  ),
                );
              },
              childCount: items.length,
            ),
          ),
        ],
      ),
    );
  }
}
